writer.go

```diff
@@ -286,24 +304,36 @@ func (p *MediaPlaylist) Remove() (err error) {
 	if p.Live {
 		p.SeqNo++
 	}
-	p.buf.Reset()
+	p.dirty = true
```

Comment:
Does this need to be protected by the lock ?

Reply:
No, it is locked in the calling function